### PR TITLE
Unify "name" and "title" field for oncall shift

### PR DESCRIPTION
**Pull request by Konstantinov-Innokentii**
# What this PR does
It drops "title" field from OnCall shift and makes "name" field not-required and nullable.  It is needed to provide ability to change shift name in web ui.

## Checklist

- [ ] Unit, integration, and e2e (if applicable) tests updated
- [ ] Documentation added (or `pr:no public docs` PR label added if not required)
- [ ] `CHANGELOG.md` updated (or `pr:no changelog` PR label added if not required)

### CODE REVIEW

File: engine/apps/schedules/models/custom_on_call_shift.py [hunk 1]
```diff
@@ -172,8 +170,9 @@ class CustomOnCallShift(models.Model):
         null=True,
         default=None,
     )
-    name = models.CharField(max_length=200)
-    title = models.CharField(max_length=200, null=True, default=None)
+    name = models.CharField(max_length=200, null=True, default=None)
+    # name = models.CharField(max_length=200)
+    # title = models.CharField(max_length=200, null=True, default=None)
```
Comment (Ferril) [inline engine/apps/schedules/models/custom_on_call_shift.py:175]: Probably forgot to remove commented code

File: engine/apps/schedules/models/custom_on_call_shift.py [hunk 2]
```diff
@@ -705,10 +704,6 @@ def create_or_update_last_shift(self, data):
         instance_data["start_rotation_from_user_index"] = start_rotation_from_user_index
 
         if self.last_updated_shift is None or self.last_updated_shift.event_is_started:
-            # create new shift
```
Comment (Ferril) [inline engine/apps/schedules/models/custom_on_call_shift.py:708]: There is no need to remove this comment as it is related to the code below.
Also I think, we should set the current's shift name here or exclude the "name" from the `fields_to_remove` above, if we want shifts belonging to the same rotation to have the same name

File: engine/apps/schedules/migrations/0012_auto_20230427_0549.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            model_name='customoncallshift',
+            name='title',
+        ),
+        migrations.AlterField(
+            model_name='customoncallshift',
+            name='name',
+            field=models.CharField(default=None, max_length=200, null=True),
+        ),
+        migrations.RunPython(drop_autogenerated_web_shift_name, autogenerate_web_shift_name),
+    ]
```
Comment (Ferril) [inline engine/apps/schedules/migrations/0012_auto_20230427_0549.py:36]: It's also probably needed to remove or change the `unique_together` constraint for the field `name`

**Review by Ferril — APPROVED** [review]

**Comment by Konstantinov-Innokentii** [comment]
merged in maxim/1502-visualize-schedule-weekends branch

**Pull request closed without merging**